From 5a627472c8e4cde38d8035a1c7c347ca6a70b954 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Sun, 16 Oct 2016 07:05:34 -0500
Subject: [PATCH 2/2] twrp: fix toolbox for CM build

---
 toolbox/Android.mk | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/toolbox/Android.mk b/toolbox/Android.mk
index dc4252d..dc82cc1 100644
--- a/toolbox/Android.mk
+++ b/toolbox/Android.mk
@@ -43,28 +43,20 @@ ifeq ($(TW_USE_TOOLBOX), true)
         # These are the only toolbox tools in M. The rest are now in toybox.
         BSD_TOOLS := \
             dd \
-            du \
 
         OUR_TOOLS := \
-            df \
             iftop \
             ioctl \
-            ionice \
             log \
             ls \
-            lsof \
-            mount \
             nandread \
             newfs_msdos \
             ps \
             prlimit \
-            renice \
             sendevent \
             start \
             stop \
             top \
-            uptime \
-            watchprops \
 
     else
         ifneq (,$(filter $(PLATFORM_SDK_VERSION), 21 22))
@@ -75,8 +67,7 @@ ifeq ($(TW_USE_TOOLBOX), true)
                 cat \
                 chown \
                 cp \
-                dd \
-                du \
+                dd \ 
                 grep \
                 kill \
                 ln \
@@ -91,7 +82,6 @@ ifeq ($(TW_USE_TOOLBOX), true)
                 cat \
                 chown \
                 dd \
-                du \
                 kill \
                 ln \
                 mv \
@@ -108,7 +98,6 @@ ifeq ($(TW_USE_TOOLBOX), true)
             clear \
             cmp \
             date \
-            df \
             dmesg \
             getevent \
             hd \
@@ -117,21 +106,17 @@ ifeq ($(TW_USE_TOOLBOX), true)
             iftop \
             insmod \
             ioctl \
-            ionice \
             log \
             lsmod \
-            lsof \
             md5 \
             mkdir \
             mkswap \
-            mount \
             nandread \
             netstat \
             newfs_msdos \
             notify \
             ps \
             readlink \
-            renice \
             rmmod \
             route \
             schedtop \
@@ -142,9 +127,7 @@ ifeq ($(TW_USE_TOOLBOX), true)
             top \
             touch \
             umount \
-            uptime \
             vmstat \
-            watchprops \
             wipe
         ifneq ($(TWHAVE_SELINUX), true)
             OUR_TOOLS += ls
-- 
1.9.1

