From 34a71dfb24861b8425333e1ea6a6d60a9d9f51bd Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Thu, 22 Jun 2017 13:46:57 -0500
Subject: [PATCH] It's official for me

---
 config/common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/common.mk b/config/common.mk
index d19aef9..41ea3a3 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -216,7 +216,7 @@ ifdef VALIDUS_BUILD_EXTRA
 endif
 
 ifndef VALIDUS_BUILD_TYPE
-    VALIDUS_BUILD_TYPE := UNOFFICIAL
+    VALIDUS_BUILD_TYPE := METICULUS
 endif
 
 # Set all versions
-- 
1.9.1

