From 1ac3350f1852605aabd3773ce0cbf75023b3ff21 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Mon, 1 May 2017 10:35:50 -0500
Subject: [PATCH] 0001-MediaCodec-use-google-h264-encoder

---
 media/libstagefright/MediaCodec.cpp | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/MediaCodec.cpp b/media/libstagefright/MediaCodec.cpp
index ff5c4d4..99cfe09 100644
--- a/media/libstagefright/MediaCodec.cpp
+++ b/media/libstagefright/MediaCodec.cpp
@@ -50,6 +50,8 @@
 #include <private/android_filesystem_config.h>
 #include <utils/Log.h>
 #include <utils/Singleton.h>
+#include <cutils/properties.h>
+#include <string.h>
 
 namespace android {
 
@@ -173,7 +175,22 @@ sp<MediaCodec> MediaCodec::CreateByType(
         const sp<ALooper> &looper, const AString &mime, bool encoder, status_t *err, pid_t pid) {
     sp<MediaCodec> codec = new MediaCodec(looper, pid);
 
-    const status_t ret = codec->init(mime, true /* nameIsType */, encoder);
+    /* Meticulus:
+     * The hw video encoder does not do well with screenrecording
+     * and the google decoder does not do well with 1080p video
+     * playback. So here we are forcing the use of the google
+     * encoder for video/avc for recording While retaining the
+     * default behavior for video playback.
+     * UPDATE: Lets make this switchable.
+     */
+    status_t ret = -1;
+    //ALOGD("Meticulus: mime = %s",mime.c_str());
+    if(!strcmp(mime.c_str(),"video/avc") && encoder && property_get_bool("persist.sys.google_avc_enc",0)) {
+	ret = codec->init("OMX.google.h264.encoder",false,encoder);
+	ALOGI("persist.sys.google_avc_enc is set.");
+    }
+    else
+	ret = codec->init(mime, true /* nameIsType */, encoder);
     if (err != NULL) {
         *err = ret;
     }
@@ -185,7 +202,14 @@ sp<MediaCodec> MediaCodec::CreateByComponentName(
         const sp<ALooper> &looper, const AString &name, status_t *err, pid_t pid) {
     sp<MediaCodec> codec = new MediaCodec(looper, pid);
 
-    const status_t ret = codec->init(name, false /* nameIsType */, false /* encoder */);
+    status_t ret = -1;
+    //ALOGD("Meticulus: name = %s",name.c_str());
+    if(!strcmp(name.c_str(),"OMX.IMG.TOPAZ.VIDEO.Encoder") && property_get_bool("persist.sys.google_avc_enc",0)) {
+	ret = codec->init("OMX.google.h264.encoder",false,true);
+	ALOGI("persist.sys.google_avc_enc is set.");
+    }
+    else
+	ret = codec->init(name, false /* nameIsType */, false /* encoder */);
     if (err != NULL) {
         *err = ret;
     }
-- 
1.9.1

